### PR TITLE
dotCMS/core#20577 Create search result page

### PR DESCRIPTION
**Pull request by rjvelazco**

### CODE REVIEW

File: components/PageError.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Terminal } from '@components/PageRenderError';
 
 export const PageError = ({ title, error }: { title: string; error: string }): JSX.Element => {
     return (
-        <main className="container mx-auto px-16 flex-grow">
+        <main className="container flex flex-col flex-grow mx-auto px-16 overflow-auto">
```
Comment (fmontes) [inline components/PageError.tsx]: What's the `overflow-auto` does here?

File: components/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                    </li>
+                )}
+            </ul>
+        </>
+    );
+};
+
+const PaginationStart = (page: number, totalPages: number, limit: number): number => {
+    const halfLimit = Math.floor(limit / 2);
+    if (totalPages < limit || page <= halfLimit) {
```
Comment (fmontes) [inline components/Pagination.tsx]: Is good practice to put this conditions in a function with a name that explain what it does.

File: components/header/components/Search.tsx [hunk 3]
```diff
@@ -1,25 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useRouter } from 'next/router';
 import { SearchIcon } from './SearchIcon';
 
 export const Search = (): JSX.Element => {
     const focusOnSearch = () => document.getElementById('search').focus();
```
Comment (fmontes) [inline components/header/components/Search.tsx]: https://reactjs.org/docs/hooks-reference.html#useref

**Review by fmontes — CHANGES_REQUESTED** [review]

File: components/PageError.tsx (hunk 1 above)
Comment (rjvelazco) [reply components/PageError.tsx]: It prevents content from exceeding the container when the error is very long.

Comment (rjvelazco) [reply components/PageError.tsx]: It prevents content from exceeding the container when the error is very long.

**Pull request merged**